libnmstate/netapplier.py

```diff
@@ -333,3 +334,14 @@ def _cleanup_iface_ethernet_state_sanitize(desired_state, current_state):
             desired_state.pop('ethernet', None)
             current_state.pop('ethernet', None)
     return desired_state, current_state
+
+
+def _cleanup_iface_ipv6_link_local_addr(desired_state, current_state):
+    for state in (desired_state, current_state):
+        for address in state.get('ipv6', {}).get('address', []):
+            addr_str = '{a[ip]}/{a[prefix-length]}'.format(a=address)
+            if nm.ipv6.is_link_local_addr(addr_str):
+                state['ipv6']['address'].remove(address)
```

Comment:
Removing inside the loop from the same list we iterate on feels dangerous in any language.
Perhaps collect the addresses needed for removal using a loop and just remove them.

Reply:
Fixed.